### PR TITLE
make a kube-apiserver config struct

### PR DESCRIPTION
**Pull request by deads2k**
creates an internal kubeapiserver struct for use by the hypershift command.

I've done a couple weird decompositions for methods shared with the combined server to make the safety obvious for 3.11

/assign @sttts @mfojtik

### CODE REVIEW

File: pkg/cmd/server/apis/config/types.go [hunk 1]
```diff
@@ -1757,3 +1757,84 @@ type ServerProjectConfig struct {
...
+	ServicesNodePortRange string
+
+	// TODO this should be removable in a later release after we've completed migration
+	LegacyServiceServingCertSignerCABundle string
+
+	// UserAgentMatchingConfig controls how API calls from *voluntarily* identifying clients will be handled.  THIS DOES NOT DEFEND AGAINST MALICIOUS CLIENTS!
+	// TODO I think we should just drop this feature.
+	UserAgentMatchingConfig UserAgentMatchingConfig
+
+	ImagePolicyConfig KubeAPIServerImagePolicyConfig
```
Comment (mfojtik) [inline pkg/cmd/server/apis/config/types.go]: doesn't think belong to openshift api server?

File: pkg/cmd/openshift-apiserver/openshiftapiserver/config.go [hunk 2]
```diff
@@ -215,22 +219,20 @@ func NewOpenshiftAPIConfig(openshiftAPIServerConfig *configapi.MasterConfig) (*O
...
+			ProjectRequestMessage:              newConfig.ProjectConfig.ProjectRequestMessage,
 			ClusterQuotaMappingController:      clusterQuotaMappingController,
 			RESTMapper:                         restMapper,
 			SCCStorage:                         sccStorage,
+			ServiceAccountMethod:               newConfig.ServiceAccountOAuthGrantMethod,
 		},
 	}
-	if openshiftAPIServerConfig.OAuthConfig != nil {
-		ret.ExtraConfig.ServiceAccountMethod = openshiftAPIServerConfig.OAuthConfig.GrantConfig.ServiceAccountMethod
-	}
```
Comment (sttts) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/config.go]: why is this gone?

Comment (sttts) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/config.go]: ic, moved up

Comment (deads2k) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/config.go]: > why is this gone?

two lines up

`ServiceAccountMethod:               newConfig.ServiceAccountOAuthGrantMethod,`

https://github.com/openshift/origin/pull/20787/files/355c3e75e7c129135f9e79fa99132f0d512ee3d0#diff-6631c206d2fb778961dcfa92e2f0f7caR233

File: pkg/cmd/server/apis/config/types.go [hunk 3]
```diff
@@ -1757,3 +1757,84 @@ type ServerProjectConfig struct {
...
+	// The list of public keys is used to verify presented service account tokens.
+	// Each key is tried in order until the list is exhausted or verification succeeds.
+	// If no keys are specified, no service account authentication will be available.
+	ServiceAccountPublicKeyFiles []string
+
+	// TODO this needs to be removed.
+	APIServerArguments ExtendedArguments
+}
+
+type KubeAPIServerImagePolicyConfig struct {
```
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/cmd/server/apis/config/types.go]: > @deads2k why does kube api server need to know these?

imagepolicyadmission uses it.

File: pkg/cmd/server/apis/config/types.go (hunk 1 above)
Comment (deads2k) [reply pkg/cmd/server/apis/config/types.go]: > doesn't think belong to openshift api server?

imagepolicyadmission uses it.

**Comment by mfojtik** [comment]
/lgtm

@sttts lgtmed per slack

**Pull request merged**